### ISSUE TITLE
How to provide DIR/File permissions while extracting the package

### ISSUE THREAD

**Issue by ishanuda** [coderaiser/node-onezip#8]
Hi,
Could anyone please explain the way to use the dir/ file permissions while extracting the zip package just like in the following library:

https://github.com/maxogden/extract-zip

Thank you very much.

**Comment by coderaiser** [comment]
What kind of mode are you talking about? extract-zip uses [mode passed by options](https://github.com/maxogden/extract-zip/blob/master/index.js#L136L159). Why do you need this?

**Comment by ishanuda** [comment]
Hi,
I experienced some libraries that did not extract the files and directories in the macOS file system because of the file permission error. It could not create the required folders.
So I selected the extract-zip and using the option with 775 I could extract the package without any errors.

The problem is, that it has no extract progress and I see this library has the progress that can use to create a progress bar.
So I wanted to try this library and before using it I just wanted to know that this library will work with macOS file systems with file permission.

Thank you very much.
